cloudferrylib/os/actions/check_sql.py

```diff
@@ -13,10 +13,23 @@
 # limitations under the License.
 
 from cloudferrylib.base.action import action
+from cloudferrylib.os.compute import nova_compute
+from cloudferrylib.os.image import glance_image
+from cloudferrylib.os.storage import cinder_database
+from cloudferrylib.utils import utils as utl
+
+SQL = "SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES LIMIT 1"
 
 
 class CheckSQL(action.Action):
 
     def run(self, info=None, **kwargs):
-        self.cloud.mysql_connector.execute("SELECT TABLE_NAME FROM "
-                                           "INFORMATION_SCHEMA.TABLES LIMIT 1")
+        cfg = self.cloud.cloud_config
+        nova = self.cloud.resources[utl.COMPUTE_RESOURCE]
+        nova_conn = nova.get_db_connection()
+        nova_conn.execute(SQL)
+        cinder = self.cloud.resources[utl.STORAGE_RESOURCE]
+        cinder.connector.execute(SQL)
+        glance = self.cloud.resources[utl.IMAGE_RESOURCE]
+        glance_conn = glance.get_db_connection()
```

Comment:
It would be nice to have the same behaviour as above for Nova and Cinder (establish connection in the resource's `__init__` method). We can fix it here or maybe in separate Minor commit.